src/App/component/DashBoardComponent/CodeBaseViews.jsx

```diff
@@ -0,0 +1,191 @@
+        setCurrentProject(response.data)
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+    fetchCurrentProject()
+  }, [])
+
+  const getCommitFromGitHub = async () => {
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+      try {
+        await Axios.post(`http://localhost:9100/pvs-api/github/commits/${query}`, "",
+          { headers: { "Authorization": `${jwtToken}` } })
+        getGitHubCommitFromDB()
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+  }
+
+  const getGitHubCommitFromDB = async () => {
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+      try {
+        const response = await Axios.get(`http://localhost:9100/pvs-api/github/commits/${query}`,
+          { headers: { "Authorization": `${jwtToken}` } })
+        setCommitListData(response.data)
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+  }
+
+  useEffect(() => {
+    if (Object.keys(currentProject).length !== 0) {
+      getCommitFromGitHub()
+    }
+  }, [currentProject, prop.startMonth, prop.endMonth])
+
+  // Get commits' total additions and deletions
+  useEffect(() => {
+    // Only triger the page rendering once
+    const calculateData = async () => {
+      await Promise.all([
+        getAdditions(),
+        getDeletions()
+      ])
+    }
+    calculateData()
+  }, [commitListData, prop.startMonth, prop.endMonth])
+
+  const getAdditions = () => {
+    const { startMonth, endMonth } = prop
+
+    const chartDataset = { additions: 0 }
+    for (let month = moment(startMonth); month <= moment(endMonth); month = month.add(1, 'months')) {
+      chartDataset.additions += (commitListData.filter(commit => {
+        return moment(commit.committedDate).format("YYYY-MM") === month.format("YYYY-MM")
+      })
+        .reduce(function (additionSum, currentCommit) {
+          return additionSum + currentCommit.additions;
+        }, 0))
+    }
+
+    setAdditions(chartDataset)
+  }
+
+  const getDeletions = () => {
+    const { startMonth, endMonth } = prop
+
+    const chartDataset = { deletions: 0 }
+    for (let month = moment(startMonth); month <= moment(endMonth); month = month.add(1, 'months')) {
+      chartDataset.deletions += (commitListData.filter(commit => {
+        return moment(commit.committedDate).format("YYYY-MM") === month.format("YYYY-MM")
+      })
+        .reduce(function (deletionSum, currentCommit) {
+          return deletionSum + currentCommit.deletions;
+        }, 0))
+    }
+
+    setDeletions(chartDataset)
+  }
+
+  const setAdditions = (chartDataset) => {
+    const job = { id: {}, job: {}, views: {} }
```

Comment:
Suggest use same type default value to create this object and its fields.
```suggestion
    const job = { id: '', job: '', views: 0 }
```